test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -34,6 +40,99 @@ public void Build_honors_UseStartup_with_string()
+                .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, typeof(WebHostBuilderTests).GetTypeInfo().Assembly.FullName)
+                .UseServer(new TestServer());
+
+            var host = (WebHost)builder.Build();
+
+            Assert.Equal("1", builder.GetSetting("testhostingstartup"));
+        }
+
+        [Fact]
+        public void Build_RunsHostingStartupAssembliesBeforeApplication()
+        {
+            var startup = new StartupVerifyServiceA();
+            var builder = CreateWebHostBuilder()
+                .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, typeof(WebHostBuilderTests).GetTypeInfo().Assembly.FullName)
+                .ConfigureServices(services =>
+                {
+                    services.AddSingleton<IStartup>(startup);
+                })
+                .UseServer(new TestServer());
+
+            var host = (WebHost)builder.Build();
+
+            Assert.NotNull(startup.ServiceADescriptor);
+            Assert.NotNull(startup.ServiceA);
+        }
+
+        [Fact]
+        public void Build_ConfigureLoggingInHostingStartupWorks()
+        {
+            var builder = CreateWebHostBuilder()
+                .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, typeof(WebHostBuilderTests).GetTypeInfo().Assembly.FullName)
+                .Configure(app =>
+                {
+                    var loggerFactory = app.ApplicationServices.GetService<ILoggerFactory>();
+                    var logger = loggerFactory.CreateLogger(nameof(WebHostBuilderTests));
+                    logger.LogInformation("From startup");
+                })
+                .UseServer(new TestServer());
+
+            var host = (WebHost)builder.Build();
+            var sink = host.Services.GetRequiredService<ITestSink>();
+            Assert.True(sink.Writes.Any(w => w.State.ToString() == "From startup"));
+        }
+
+        [Fact]
+        public void Build_DoesNotRunHostingStartupAssembliesDoNotRunIfNotSpecified()
+        {
+            var builder = CreateWebHostBuilder()
+                .UseServer(new TestServer());
+
+            var host = (WebHost)builder.Build();
+
+            Assert.Null(builder.GetSetting("testhostingstartup"));
+        }
+
+        [Fact]
+        public void Build_ThrowsIfUnloadableAssemblyNameInHostingStartupAssemblies()
+        {
+            var builder = CreateWebHostBuilder()
+                .CaptureStartupErrors(false)
+                .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, "SomeBogusName")
+                .UseServer(new TestServer());
+
+            Assert.Throws<FileNotFoundException>(() => (WebHost)builder.Build());
+        }
+
+        [Fact]
+        public void Build_DoesNotThrowIfUnloadableAssemblyNameInHostingStartupAssembliesAndCaptureStartupErrorsTrue()
+        {
+            var builder = CreateWebHostBuilder()
+                .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, "SomeBogusName")
```

Comment:
Use an explicit `.CaptureStartupErrors(true)`.

Is there a test that verifies the default is `true`?